clippy_lints/src/methods/redundant_repeat_take.rs

```diff
@@ -0,0 +1,32 @@
+use clippy_utils::diagnostics::span_lint_and_sugg;
+use clippy_utils::source::snippet;
+use rustc_errors::Applicability;
+use rustc_hir::{Expr, ExprKind, QPath};
+use rustc_lint::LateContext;
+
+use super::REDUNDANT_REPEAT_TAKE;
+
+pub(super) fn check(cx: &LateContext<'_>, expr: &Expr<'_>, repeat_expr: &Expr<'_>, take_arg: &Expr<'_>) {
+    if !expr.span.from_expansion()
+        && let ExprKind::Call(inner, repeat_args) = repeat_expr.kind
+        && repeat_args.len() == 1
+        && let ExprKind::Path(ref qpath) = inner.kind
+        && let QPath::Resolved(_, path) = qpath
+        && path.segments.len() == 1
+        && path.segments[0].ident.name.as_str() == "repeat"
```

Comment:
`iter::repeat` is a diagnostic item, so you should be able to match against its `DefId` rather than looking for exact paths. Something like
```rs
   && let Some(def_id) = fn_def_id(inner)
   && cx.tcx.is_diagnostic_item(sym::iter_repeat, def_id)
```

Reply:
There should also be a check that the `.take()` call is the `take` from the Iterator trait and not from a user defined trait, with `is_trait_method(cx, expr, sym::Iterator)`